impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultSourceRoot.java

```diff
@@ -79,7 +79,12 @@ public DefaultSourceRoot(final Session session, final Path baseDir, final Source
         if (value != null) {
             directory = baseDir.resolve(value);
         } else {
-            directory = baseDir.resolve("src").resolve(scope.id()).resolve(language.id());
+            var src = baseDir.resolve("src");
```

Comment:
Do we use `var` in maven codebase? I kinda remember we agreed not to? (but I may be missing some later agreement).

Reply:
Right, it was my mistake.

Personally, I sometime use `var` in expressions such as `var x = new Foo()` when there is no way for the type to be something else than what we can read on the line, and usually refrain to use it in other contexts. I do not know if there is a Maven policy.

Reply:
I strongly prefer to avoid var. IMHO it should never have been added to the language. I want to know the types of things when I'm reading the code. var hides that info. I also don't want method return types to change without explicit consideration and inspection of every call point.